src/quicer.erl

```diff
@@ -200,9 +200,28 @@ listen(ListenOn, Opts) when is_map(Opts) ->
   quicer_nif:listen(ListenOn, Opts).
 
 %% @doc close listener with listener handle
--spec close_listener(listener_handle()) -> ok.
+-spec close_listener(listener_handle()) -> ok | {error, badarg | closed | timeout}.
 close_listener(Listener) ->
-  quicer_nif:close_listener(Listener).
+  close_listener(Listener, 5000).
+
+-spec close_listener(listener_handle(), timer:time()) ->
+        ok | {error, badarg | closed | timeout}.
+close_listener(Listener, Timeout) ->
+  case quicer_nif:close_listener(Listener) of
+    ok ->
+      receive
+        {quic, listener_stopped, Listener} ->
```

Comment:
The event is only shot to the Listener owner who opens the listener. 
The timeout prevents long blocking on the caller who is not the owner.
maybe we should check the owner ship in NIF and return `{error, not_owner}`